### ISSUE TITLE
Add checksum hints to API

### ISSUE THREAD

**Issue by dcramer** [getsentry/sentry#1657]
Currently we have a 'checksum' attribute in the API. We want to deprecate that (as people shouldnt need to know how the server works), and add a new attribute which allows you to send both free-standing and templated hints.

```
# a completely custom fingerprint
fingerprint = ['my', 'custom', 'values']

# appending to the default value
fingerprint = ['{{default}}', 'my', 'value']

# FUTURE USE: adding additional templated params
fingerprint = ['{{default}}', '{{http.url}}', '{{http.method}}']

# client use
client.captureException(fingerprint=[...])
```

**Comment by mitsuhiko** [comment]
To make future use work should we error to the clients if they send `]{]{[a-zA-Z0-9._]+]}]}`?

**Comment by dcramer** [comment]
@mitsuhiko never error -- but we can store the hints and store errors attached to the group (though that'd be a "nice to have")

**Comment by mitsuhiko** [comment]
So just ignore an item that looks like a template.

**Comment by dcramer** [comment]
I think we'd say "if you insert a value that represents a template, and it doesnt match to a supported attribute, it effectively translates to a blank value"

**Comment by dcramer** [comment]
Available in Sentry 7.7.0
